shared/reducers/chat2.js

```diff
@@ -245,17 +242,23 @@ const messageMapReducer = (messageMap, action, pendingOutboxToOrdinal) => {
         const path = action.error ? '' : action.payload.path
         return message.set('downloadPath', path)
       })
+    case Chat2Gen.metasReceived:
+      return action.payload.clearExistingMessages
+        ? messageMap.filter((_, k) => k !== Constants.pendingConversationIDKey)
```

Comment:
likely faster to just pull it out and add it back in instead of going over all the keys
```
const existing = messageMap.get(Constants.pendingConversationIDKey)
messageMap.clear().add(Constants.pendingConversationIDKey, existing)
```